compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -189,13 +190,23 @@ public void save() {
     private void writeDependencies() {
         Package currentPackage = this.currentPackage();
         if (currentPackage != null) {
-            ResolvedPackageDependency resolvedPackageDependency =
-                    new ResolvedPackageDependency(currentPackage, PackageDependencyScope.DEFAULT);
+            Comparator<ResolvedPackageDependency> comparator = (o1, o2) -> {
+                if (o1.packageInstance().packageOrg().equals(o2.packageInstance().packageOrg())) {
+                    return o1.packageInstance().packageName().toString()
+                            .compareTo(o2.packageInstance().packageName().toString());
+                }
+                return o1.packageInstance().packageOrg().toString()
+                        .compareTo(o2.packageInstance().packageOrg().toString());
+            };
+
             Collection<ResolvedPackageDependency> pkgDependencies =
-                    currentPackage.getResolution().dependencyGraph().getDirectDependencies(resolvedPackageDependency);
+                    currentPackage.getResolution().dependencyGraph().getNodes().stream().filter(resolvedDep ->
```

Comment:
Here we need to get package dependencies along with their transitive dependencies to add to the `Dependencies.toml` right? Can we get transitive dependencies of a given `graphDependency`?

Reply:
Yes, we can modify it once the dependency toml is fixed to support table of arrays